packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -182,6 +182,14 @@ export function updateWrapper(element: Element, props: Object) {
     } else if (node.value !== '' + value) {
       node.value = '' + value;
     }
+  } else if (
+    (props.type === 'submit' || props.type === 'reset') &&
+    (value === null || value === undefined)
```

Comment:
This condition seems unnecessary because we're in an `else` branch for `if (value != null)`.